### ISSUE TITLE
[Bug]: Sync Connection Timeout

### ISSUE THREAD

**Issue by teclab-at** [nextcloud/desktop#7184]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

The desktop client stops syncing with error "Connection Timeout".

Server is a fresh install via "Web installer" running on Ubuntu 22.

Logs:

[nextcloud_timeout.zip](https://github.com/user-attachments/files/17091002/nextcloud_timeout.zip)

Screenshot

![Untitled](https://github.com/user-attachments/assets/6995b5d1-eb8f-44eb-90c1-ec970bf9bf1b)

### Steps to reproduce

- I have a data-set on the server of about 23GB +12000 Files.
- Login with the desktop client, select the folders to sync, enable "Virtual File Support" and all files get synced.
- Then select the root folder and enable "Always keep on this device". Now the sync starts, but stops occasionally on files with error "Connection Timeout".
- A Force Sync continous syncing but stops on other files.
- Limit Download Bandwith to "Limit automatically" (at least it seems so) runs and finishes without error.

### Expected behavior

Sync finishes without error.

### Which files are affected by this bug

IMG_20181111_143840663.jpg

### Operating system

Windows

### Which version of the operating system you are running.

Windows 10

### Package

Other

### Nextcloud Server version

30.0.0

### Nextcloud Desktop Client version

3.14.0

### Is this bug present after an update or on a fresh install?

Fresh desktop client install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
{"reqId":"T3fLulVmt2inN1QnGkjL","level":0,"time":"2024-09-22T19:33:29+00:00","remoteAddr":"10.10.10.133","user":"thomas","app":"no app in context","method":"GET","url":"/remote.php/dav/files/thomas/Zertifikate/SIEMENS/Secure%20throughout%20the%20day%20(SEC3).pdf","message":"Calling OCP\\DB\\QueryBuilder\\IQueryBuilder::orX without parameters is deprecated and will throw soon.","userAgent":"Mozilla/5.0 (Windows) mirall/3.14.0stable-Win64 (build 20240914) (Nextcloud, windows-10.0.19045 ClientArchitecture: x86_64 OsArchitecture: x86_64)","version":"30.0.0.14","exception":{"Exception":"Exception","Message":"No parameters in call to OC\\DB\\QueryBuilder\\ExpressionBuilder\\ExpressionBuilder::orX","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/circles/lib/Tools/Db/ExtendedQueryBuilder.php","line":488,"function":"orX","class":"OC\\DB\\QueryBuilder\\ExpressionBuilder\\ExpressionBuilder","type":"->"},{"file":"/var/www/nextcloud/apps/circles/lib/Tools/Db/ExtendedQueryBuilder.php","line":325,"function":"exprLimitNull","class":"OCA\\Circles\\Tools\\Db\\ExtendedQueryBuilder","type":"->"},{"file":"/var/www/nextcloud/apps/circles/lib/Db/ShareWrapperRequest.php","line":312,"function":"limitNull","class":"OCA\\Circles\\Tools\\Db\\ExtendedQueryBuilder","type":"->"},{"file":"/var/www/nextcloud/apps/circles/lib/Service/ShareWrapperService.php","line":225,"function":"getSharedWith","class":"OCA\\Circles\\Db\\ShareWrapperRequest","type":"->"},{"file":"/var/www/nextcloud/apps/circles/lib/ShareByCircleProvider.php","line":540,"function":"getSharedWith","class":"OCA\\Circles\\Service\\ShareWrapperService","type":"->"},{"file":"/var/www/nextcloud/lib/private/Share20/Manager.php","line":1204,"function":"getSharedWith","class":"OCA\\Circles\\ShareByCircleProvider","type":"->"},{"file":"/var/www/nextcloud/apps/files_sharing/lib/MountProvider.php","line":47,"function":"getSharedWith","class":"OC\\Share20\\Manager","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php","line":72,"function":"getMountsForUser","class":"OCA\\Files_Sharing\\MountProvider","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Config/MountProviderCollection.php","line":129,"function":"getMountsFromProvider","class":"OC\\Files\\Config\\MountProviderCollection","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/SetupManager.php","line":207,"function":"addMountForUser","class":"OC\\Files\\Config\\MountProviderCollection","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/SetupManager.php","line":306,"function":"OC\\Files\\{closure}","class":"OC\\Files\\SetupManager","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/lib/private/Files/SetupManager.php","line":208,"function":"setupForUserWith","class":"OC\\Files\\SetupManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/SetupManager.php","line":380,"function":"setupForUser","class":"OC\\Files\\SetupManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Mount/Manager.php","line":75,"function":"setupForPath","class":"OC\\Files\\SetupManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":1344,"function":"find","class":"OC\\Files\\Mount\\Manager","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/Directory.php","line":177,"function":"getFileInfo","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php","line":95,"function":"getChild","class":"OCA\\DAV\\Connector\\Sabre\\Directory","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/DAV/ViewOnlyPlugin.php","line":63,"function":"getNodeForPath","class":"Sabre\\DAV\\Tree","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"checkViewOnly","class":"OCA\\DAV\\DAV\\ViewOnlyPlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Server.php","line":370,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/appinfo/v2/remote.php","line":19,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/remote.php","line":146,"args":["/var/www/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/nextcloud/lib/private/DB/QueryBuilder/ExpressionBuilder/ExpressionBuilder.php","Line":87,"message":"Calling OCP\\DB\\QueryBuilder\\IQueryBuilder::orX without parameters is deprecated and will throw soon.","exception":{},"CustomMessage":"Calling OCP\\DB\\QueryBuilder\\IQueryBuilder::orX without parameters is deprecated and will throw soon."}}
```

**Comment by UH-Nerion** [comment]
Same issue here. On our side it's happening when sync a pdf.
But in the log it's exactly the same issue 

 Exception No parameters in call to OC\DB\QueryBuilder\ExpressionBuilder\ExpressionBuilder::orX
Calling OCP\DB\QueryBuilder\IQueryBuilder::orX without parameters is deprecated and will throw soon.

**Comment by mgallien** [comment]
@teclab-at 
@UH-Nerion 
I am trying to investigate
do you have any UI visible when getting errors with the timeout of network connections ?

**Comment by GiambaFr** [comment]
Same for me, with or without the app window open. Win 10 on a 100Mb/s connection.
For my other computer, WIN 11, WIFI 5Ghz, all seems OK.
Same versions.

**Comment by teclab-at** [comment]
I left the Settings window open, but the error happens even when it is closed.

**Comment by mgallien** [comment]
@teclab-at @GiambaFr 
I have a custom build with a possible work around for this bug
could you test it ?
https://cloud.nextcloud.com/s/cf4iBmPHxqdi7jZ
please keep in mind that this build just got tested for this particular issue not for general QA

**Comment by GiambaFr** [comment]
@mgallien
Ok, Just tested, 6GB transfered without any error.
Can't wait for a new mainline version.
Thanks

**Comment by elr3m** [comment]
Same behavior here (connection timeout and operation cancelled) on Windows 11.
I used @mgallien installer and it seem to have fixed it, I even unchecked and re-checked 22Gb of data and it all downloaded.

**Comment by UH-Nerion** [comment]
I installed your version, but same issue here.
Problem is not resolved.

**Comment by mgallien** [comment]
@UH-Nerion 
what is the exact issue you get ?
could you share logs or screenshots ?

**Comment by UH-Nerion** [comment]
After 3-4 minutes it was working fine with your new release and the full sync was finished without any more errors.
Don't know why it was not working directly at the beginning.

**Comment by mgallien** [comment]
this is now solved with the release of 3.14.1

**Comment by GiambaFr** [comment]
Your test version was 3.14.50 isn't it?

**Comment by Gwindalmir** [comment]
I'd like to comment that while this did fix the syncing error on two of my systems, one of them (Framework 13) is showing more connection timeout errors after resuming from sleep (even tried toggling wifi after resume).
This is all while I can browse my nextcloud instance in Firefox just fine at the same time.
I had to restart the desktop client for it to resume. It's like it cached the error state and didn't actually retry.

I've attached the logs files from that session. I changed one of the last ones to remove the list of filenames it was syncing once it reconnected successfully.
[logs.zip](https://github.com/user-attachments/files/17183054/logs.zip)
